### DISCUSSION TITLE
SSH PASSWORD rotation

### DISCUSSION THREAD

**Discussion by chuckkay** [X1Plus/X1Plus#83] [Q&A]
i changed the provide ssh password , to my own by logging in with provided password, then sudo - then passwd 
My question what if i accidently his the refresh ssh password on the touch screen, after already creating my own.

**Comment by jwise** [comment]
The password is always reset to the screen password on boot.  If you want to persist login credentials, use SSH keys :)

**Reply by mattcaron** [reply]
Can confirm SSH keys work... Now I'm just trying to find out where the `dropbear` options are specified so I can pass it `-s` to not allow password auth.

**Reply by mattcaron** [reply]
Oh, I should probably mention:

- if you want to connect as `root`, create `/root/.ssh/authorized_keys` (which is also `/config/sshd/authorized_keys` because of symlink fun)
- if you want to connect as `bblp`:
  1. create `/mnt/sdcard/.ssh/authorized_keys`
  2. edit `/etc/passwd` and set `bblp`'s shell to `/bin/sh` (otherwise it just punts you off)

**Reply by jwise** [reply]
Yeah, this is correct.  I recommend not connecting as `bblp`.

We have the symlink to `/config/sshd` because `/root/.ssh` is part of the read-write overlay partition that gets wiped on X1Plus reinstalls (unless you explicitly persist it... not recommended).

**Reply by mattcaron** [reply]
Can I ask why you recommend not connecting as `bblp`? I was thinking that a UID 1000 user is better than UID 0. Plus the homedir is `/mnt/sdcard`, so it seemed appropriate. But, changing the shell from `/bin/false` did give me pause. What am I not seeing?

**Reply by jwise** [reply]
All the daemons on the system run as root anyway, and process untrusted content with reckless abandon.  You can do very little that is useful as `bblp`, and attempts to make `bblp` more useful essentially devolve into making `bblp` the same as `root`.

**Reply by mattcaron** [reply]
I did 10+ years in industrial automation, including for secure environments. My eye is twitching a little bit. Lol.

(In all fairness, I have the same reaction to buildroot. We hated it so much, we (mostly me) built our own.)

**Reply by jwise** [reply]
Yeah, I know...  It is an interesting printer but it is definitely not a hardened piece of infrastructure.

**Reply by mattcaron** [reply]
And this is part of why I've never trusted it. It's not a benign device - it can make things catch on fire. There is a degree of trust because I assumed that the stock firmware actually turned off the WiFi when I told it to (I can also check the router, and it seemed to go away when I told it to), and I'd typically just run it disconnected except when checking for new firmware. But it could be lying... I'd just move stuff with Micro SD cards. But I hate Micro SD cards. Hence why I'm pushing on https://github.com/X1Plus/X1Plus/discussions/523. Because then I'll just leave WiFi off all the time, and I'll just plug in Ethernet when a new X1Plus version drops, upgrade, then unplug it, and use USB sticks for actually printing things.

Put another way, exposure surface has a time component too. If you can shorten that time, it helps.

As an aside, last night I read https://github.com/X1Plus/X1Plus?tab=readme-ov-file#how-does-x1plus-work-from-the-printers-perspective. You (plural, meaning the team) are insane, in all the best ways. I didn't realize you were doing what you were doing. Very impressive reverse engineering / monkeypatching. Much respect.
